### PR TITLE
[pkg/stanza/fileconsumer] Refactor makeReaders into makeReader

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** Just some code refactoring and to separate these changes from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19448/

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
I don't think we can accept a worker pool unless it is behind a featuregate. Can you please remove the worker related changes from this PR?

**Comment by VihasMakwana** [comment]
Okay sounds fair

File: pkg/stanza/fileconsumer/file.go [hunk 1]
```diff
@@ -182,82 +192,67 @@ func (m *Manager) consume(ctx context.Context, paths []string) {
 	m.roller.roll(ctx, readers)
 	m.saveCurrent(readers)
 	m.syncLastPollFiles(ctx)
+	m.clearCurrentFingerprints()
 }
 
-// makeReaders takes a list of paths, then creates readers from each of those paths,
+// makeReaders takes a lis of path, then creates reader,
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/file.go:198]: Typos in this comment, but more importantly, please review the entire comment as it no longer appears to describe the function.

File: pkg/stanza/fileconsumer/file.go [hunk 2]
```diff
@@ -182,82 +192,67 @@ func (m *Manager) consume(ctx context.Context, paths []string) {
...
-		files = append(files, file)
+		m.seenPaths[path] = struct{}{}
+	}
+	file, err := os.Open(path) // #nosec - operator must read in files defined by user
+	if err != nil {
+		m.Errorf("Failed to open file", zap.Error(err))
+		return nil
 	}
 
 	// Get fingerprints for each file
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/file.go:217]: This comment can be removed

File: pkg/stanza/fileconsumer/file.go [hunk 3]
```diff
@@ -182,82 +192,67 @@ func (m *Manager) consume(ctx context.Context, paths []string) {
...
-				i--
-				continue OUTER
+	for i := 0; i < len(m.currentFps)-1; i++ {
+		fp2 := m.currentFps[i]
+		if fp.StartsWith(fp2) || fp2.StartsWith(fp) {
+			// Exclude duplicates
+			if err = file.Close(); err != nil {
+				m.Errorf("problem closing file", "file", file.Name())
 			}
+			m.currentFps = m.currentFps[:len(m.currentFps)-1]
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/file.go:241]: We're adding this just above, then having to adjust the iteration bounds to account for it, then possibly removing it. Why not just add it after we've determined we actually want to keep it?

File: pkg/stanza/fileconsumer/file.go (hunk 1 above)
Comment (VihasMakwana) [reply pkg/stanza/fileconsumer/file.go]: does it look ok now?

File: pkg/stanza/fileconsumer/file.go (hunk 3 above)
Comment (VihasMakwana) [reply pkg/stanza/fileconsumer/file.go]: Addressed this change

**Review by djaglowski — APPROVED** [review]
Thanks @VihasMakwana

**Pull request merged**